theano/scan_module/scan_op.py

```diff
@@ -2564,7 +2573,19 @@ def compute_all_gradients(known_grads):
             zip(outputs[offset:offset + self.n_seqs],
                 type_outs[offset:offset + self.n_seqs])):
             if t == 'connected':
-                gradients.append(x[::-1])
+                # If the forward scan is in as_while mode, we need to pad
+                # the gradients, so that they match the size of the input
+                # sequences.
+                if self.as_while:
+                    n_zeros = inputs[0] - n_steps
+                    shp = (n_zeros,)
+                    if x.ndim > 1:
```

Comment:
The if is useless. If it is of ndim 0 or 1, you would concatenate an empty tuple to shp. So the answer would be good. No harm to let it here.